ksql-benchmark/README.md

```diff
@@ -0,0 +1,78 @@
+# KSQL JMH Microbenchmarks
+
+This module is for JMH micro-benchmarking of pieces of the KSQL code.
+
+## `SerdeBenchmark.java`
+
+For example, `SerdeBenchmark.java`
+benchmarks the performance of the Avro and JSON serdes used by KSQL, since the serdes have been
+shown to be a performance bottleneck in the past. The benchmarks use the schema files found in
+`src/main/resources/schemas`. A serialization and deserialization benchmark is run for each schema
+(e.g., `impressions` or `metrics`) and each serialization format (Avro or JSON).  
+
+### How to run
+
+The benchmarks can be run either from `SerdeBenchmark.java` directly through IntelliJ, or via the
+command line as follows, after building the module to produce `target/benchmarks.jar`:
+
+```
+java -jar ./target/benchmarks.jar
+```
+
+### Running a subset of benchmarks
+
+To run only a subset of the benchmarks, you can specify parameters to run with. For example,
+to run only Avro benchmarks:
+```
+java -jar ./target/benchmarks.jar -p serializationFormat=Avro
+```
+
+Or to run only JSON (serialization and deserialization) benchmarks using the `metrics` schema:
+```
+java -jar ./target/benchmarks.jar -p serializationFormat=JSON -p schemaName=metrics
+```
+
+Or to run only the deserialization benchmarks on both the `impressions` and `metrics` schemas:
+```
+java -jar ./target/benchmarks.jar SerdeBenchmark.deserialize -p schemaName=impressions,metrics
+```
+
+### Running with non-default parameters
+
+JMH parameters of interest may include the number of forks to use (`-f`), the number of warmup and
```

Comment:
I think it might be good to link to https://openjdk.java.net/projects/code-tools/jmh/

Reply:
Is there a particular section of those docs you think is most relevant? All the main page has to say about command line options is

> Running the benchmarks. After the build is done, you will get the self-contained executable JAR, which holds your benchmark, and all essential JMH infrastructure code:
> $ java -jar target/benchmarks.jar
> Run with -h to see the command line options available.

which  is already captured in the README.

Reply:
Not specifically, but that page just has a bunch of really good descriptions on _why_ you might want to change those parameters (e.g. https://hg.openjdk.java.net/code-tools/jmh/file/5984e353dca7/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_12_Forking.java)

Feel free to ignore if you think it just adds verbosity.

Reply:
Got'cha. I'll pass on this one since I find it difficult to find specific information starting from the JMH homepage and prefer googling, e.g., "JMH parameter forks" directly if looking for an explanation instead. Thanks though!